packages/botfuel-dialog/src/adapters/adapter.js

```diff
@@ -22,11 +28,9 @@ const MissingImplementationError = require('../errors/missing-implementation-err
  * An adapter adapts the messages to the messaging platform.
  */
 class Adapter {
-  /**
-   * @constructor
-   * @param {Object} bot - the bot
-   */
-  constructor(bot) {
+  bot: Bot;
```

Comment:
I don't know Flow but why do we have to declare the type twice?

Reply:
You have to define the types of class members (`bot: Bot;`).

The constructor takes a parameter `bot` and puts it in `this.bot`. Flow does not infer the types of method parameters from how they are used. It is then necessary to annotate the parameter type.

It seems redundant but I think both are necessary:
* the parameter annotation documents the signature of the constructor. A user should not need to read the source code of the constructor to deduce that it takes a `Bot`.
* the class member annotation is needed for each time `this.bot` is used inside the class.